Go code failing to lint

We are seeing CI failures with the go code failing to lint.

https://github.com/dask/dask-kubernetes/actions/runs/7453335524/job/20278478301?pr=820

```
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.3
go: downloading k8s.io/api v0.24.1
go: downloading k8s.io/client-go v0.24.1
go: downloading k8s.io/apimachinery v0.24.1
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading k8s.io/klog/v2 v2.90.1
go: downloading github.com/google/gofuzz v1.1.0
go: downloading github.com/golang/protobuf v1.5.3
go: downloading github.com/google/gnostic v0.5.7-v3refs
go: downloading golang.org/x/net v0.17.0
go: downloading k8s.io/utils v0.0.0-20230209194617-a36077c30491
go: downloading k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f
go: downloading golang.org/x/time v0.0.0-20220210224613-90d013bbcef8
go: downloading github.com/evanphx/json-patch v4.12.0+incompatible
go: downloading github.com/json-iterator/go v1.1.12
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading github.com/go-logr/logr v1.2.3
go: downloading google.golang.org/protobuf v1.28.1
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading golang.org/x/term v0.13.0
go: downloading golang.org/x/oauth2 v0.0.0-20220223155221-ee480838109b
go: downloading github.com/google/go-cmp v0.5.9
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
go: downloading golang.org/x/sys v0.13.0
go: downloading golang.org/x/text v0.13.0
go: downloading github.com/emicklei/go-restful/v3 v3.9.0
go: downloading github.com/go-openapi/swag v0.22.3
go: downloading github.com/go-openapi/jsonreference v0.20.1
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/go-openapi/jsonpointer v0.19.6
go: downloading github.com/josharian/intern v1.0.0
dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/utils.go:6:2: no required module provides package github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1; to add it:
	go get github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1
dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/utils.go:7:2: no required module provides package github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1; to add it:
	go get github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1
```

--- a/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1/daskclusterstatus.go
+++ b/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1/daskclusterstatus.go
@@ -3,7 +3,7 @@
 package v1
 
 import (
-	v1 "github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1"
+	v1 "github.com/dask/dask-kubernetes/v2024/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1"
 )
 
 // DaskClusterStatusApplyConfiguration represents an declarative configuration of the DaskClusterStatus type for use

--- a/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1/daskjobstatus.go
+++ b/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1/daskjobstatus.go
@@ -3,7 +3,7 @@
 package v1
 
 import (
-	kubernetesdaskorgv1 "github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1"
+	kubernetesdaskorgv1 "github.com/dask/dask-kubernetes/v2024/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 

--- a/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/utils.go
+++ b/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/utils.go
@@ -3,8 +3,8 @@
 package applyconfiguration
 
 import (
-	v1 "github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1"
-	kubernetesdaskorgv1 "github.com/dask/dask-kubernetes/v2023/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1"
+	v1 "github.com/dask/dask-kubernetes/v2024/dask_kubernetes/operator/go_client/pkg/apis/kubernetes.dask.org/v1"
+	kubernetesdaskorgv1 "github.com/dask/dask-kubernetes/v2024/dask_kubernetes/operator/go_client/pkg/client/applyconfiguration/kubernetes.dask.org/v1"
 	schema "k8s.io/apimachinery/pkg/runtime/schema"
 )
 

